rudder-core/src/main/scala/com/normation/cfclerk/xmlparsers/TechniqueParser.scala

```diff
@@ -312,13 +320,54 @@ class TechniqueParser(
    * @return A compatible variable corresponding to the entry xml
    */
   def parseCompatibleTag(xml: Node): Compatible = {
-    if(xml.label != COMPAT_TAG) throw new ParsingException("CompatibleParser was expecting a <%s> xml and get:\n%s".format(COMPAT_TAG, xml))
+    if(xml.label != COMPAT_TAG) throw new ParsingException("CompatibleParser was expecting a <%s> xml and got:\n%s".format(COMPAT_TAG, xml))
     val os = xml \ COMPAT_OS map (n =>
       OperatingSystem(n.text, (n \ "@version").text))
     val agents = xml \ COMPAT_AGENT map (n =>
       Agent(n.text, (n \ "@version").text))
     Compatible(os, agents)
   }
+
+  /* parse RUNHOOKS xml node, which look like that:
+    <RUNHOOKS>
+      <PRE name="package">
+        <PARAMETER name="package" value="vim"/>
+      </PRE>
+      <POST name="servive" if="debian">
+        <PARAMETER name="service" value="some value"/>
+        <PARAMETER name="a post command">/something/that/is/complicated "with" 'all sort of quote'</PARAMETER>
+      </POST>
+    </RUNHOOKS>
+  */
+  def parseRunHooks(id: TechniqueId, xml: Node): List[RunHook] = {
+    def parseOneHook(xml: Node, kind: RunHook.Kind): RunHook = {
+      def opt(s: String) = if(s == null || s == "") None else Some(s)
+
+      (xml \ "@name").text match {
+        case null | "" => throw new ParsingException(s"Error: in technique '${id.toString()}', tried to parse a run hooks, but attribute 'name' is missing in: ${xml}")
+        case name      =>
+          RunHook(name, kind, (xml \ "@condition").text, (xml \\ "PARAMETER").toList.flatMap(p =>
+            for {
+              pname  <- opt((p \ "@name" ).text)
+                        // for value, look first for <PARAMETER ... value="pvalue"/> and then <PARAMETER>pvalue</PARAMETER>
+              pvalue <- opt((p \ "@value").text) orElse opt(p.text)
+            } yield {
+              RunHook.Parameter(pname, pvalue)
+            }
+         ))
+      }
+    }
+
+    if(xml.label != RUN_HOOKS) throw new ParsingException(s"Error: in technique '${id.toString()}', tried to parse a <${RUN_HOOKS}> xml, but actually got: ${xml}")
```

Comment:
Do you think those checks are usefull if before calling that function we do "xml \ RUN_HOOKS"

The same goes for all other labels ...

Reply:
I hope one day we will have time to remove theose exception and put the result in an Either[ParsingError,Result]

Reply:
They are, because without them the function is not autonomous. And they were added because at some point, we bang our head on a bug to discover that the node wasn't the correct anymore following a refactoring...
They may exist better way (less boilerplate) to do so with some sugar syntax

Reply:
Developper life is not the same if we cannot bang our head on bugs ... ;) (So great, keep theose checks !!!)